state-chain/chains/src/sol/instruction_builder.rs

```diff
@@ -48,6 +46,17 @@ pub struct SolanaInstructionBuilder;
 const COMPUTE_LIMIT: SolComputeLimit = 300_000u32;
 
 impl SolanaInstructionBuilder {
+	// TODO: Shares code with EVM's ExecutexSwapAndCall or make it a function of ForeignChainAddress
+	fn destructure_address(source_address: Option<ForeignChainAddress>) -> Vec<u8> {
```

Comment:
This is also used in the EVM's ExecutexSwapAndCall. Should we have some common code or maybe just have it as a function of the ForeignChainAddress struct? @dandanlen @ramizhasan111

Reply:
We can already do this using the ```to_encoded_address``` function implemented for the ```ForeignChainAddress``` type. We should also probably also use that function for the ```destructure_address``` in evm's ExecutexSwapAndCall.

Reply:
I agree, adding this additional struct method seems like overkill.

Reply:
Seems like unfortunately that doesn't work (Ramiz has also checked). So unless someone opposes to it I'll implement an extra struct method which we can reuse when egressing CCM calls for any chain.